spec/manuals_publisher/update_published_content_spec.rb

```diff
@@ -0,0 +1,46 @@
+feature "Updating content on Manuals Publisher", manuals_publisher: true do
+  include ManualsHelpers
+
+  let(:title) { title_with_timestamp }
+  let(:section_title) { title_with_timestamp }
+  let(:change_note) { sentence }
+
+  scenario "Update a published manual with a new section" do
+    given_there_is_a_published_manual_with_a_section
+    when_i_add_a_new_section_with_a_change_note
+    and_i_publish_the_manual
+    then_the_manual_contents_page_links_to_the_new_section_on_gov_uk
+    and_the_update_log_has_the_change_note
+  end
+
+  def given_there_is_a_published_manual_with_a_section
+    create_draft_manual(title: title)
+    create_manual_section
+    publish_manual
+  end
+
+  def when_i_add_a_new_section_with_a_change_note
+    create_manual_section(title: section_title, change_note: change_note)
+  end
+
+  def and_i_publish_the_manual
+    publish_manual
+  end
+
+  def then_the_manual_contents_page_links_to_the_new_section_on_gov_uk
+    url = find_link("View on website")[:href]
+    reload_url_until_status_code(url, 200)
```

Comment:
As this is testing an update I wonder if this is vulnerable to failing in situations where it browses to this file but it's still showing the one that was first published? 

We do something of checking that a page has updated text here: https://github.com/alphagov/publishing-e2e-tests/blob/master/spec/specialist_publisher/change_note_spec.rb#L53 to check that the page that is 200ing is the updated page (it's admittedly a bit grim but not sure of a better approach)

Reply:
Good spot!